### ISSUE TITLE
Bug when using errorDisplay.close()

### ISSUE THREAD

**Issue by qbadev** [videojs/video.js#5509]
## Description
In my case, after getting error code 4, I need to replace error modal with my own modal object. So I use `player.errorDisplay.close()`, where `player` is an object returned by `videojs` function. For a test I use 7.2.3 version, but the same problem occurs in previous versions (5.20.5 from bower in my particular example).

## Steps to reproduce
1. Create an html page. Include newest videojs from cdn. Place a \<video\> object with a \<source\> with `src` to a non-existing file or a media file with not accepted codec.
2. Add `error` event handling to player and place `player.errorDisplay.close()` inside triggered function.
```
  <video id="vp" class="video-js vjs-default-skin" controls>
    <source src="sample.mp5" type="video/mp4">
  </video>
  <script type="text/javascript">
    var player = videojs('vp', {}, function(){
      this.on('error', function(){
        player.errorDisplay.close();
      });
    });
  </script>
```
3. Open a page in a browser.
3. Open Developer Tools and see any errors output in Console window.

### Expected
Error modal should disappear without any errors.

### Actual
Error modal does not disappear. Error appears in Console in Developer Tools.

### Error output
**From WebKit browsers**
Uncaught TypeError: Cannot read property 'nodeName' of undefined
at validateTarget (VM3185 video.js:2464)
at normalizeListenArgs (VM3185 video.js:2542)
at Player.on$$1 [as on] (VM3185 video.js:2616)
at Player.addTechControlsListeners_ (VM3185 video.js:22549)
at Player.controls (VM3185 video.js:24679)
at ErrorDisplay.close (VM3185 video.js:5550)
at <anonymous>:1:21

**From Firefox**
TypeError: target is undefined - debugger eval code:2464:9

#### videojs
7.2.3
#### browsers
Iron (on Chrome) 66.0.3450.0
Chrome  61.0.3163.100
Firefox 62.0.3
#### OSes
Ubuntu Linux 18.04 x64

**Comment by misteroneill** [comment]
I couldn't reproduce this issue. I tried creating a [reduced test case](https://jsbin.com/hucuwir/edit?html,console,output), but it works fine.

**Comment by misteroneill** [comment]
The `player` variable and `this` (inside the ready callback) are the same object; so, that has no bearing on this. Your example:

```
  <video id="my_video_1" class="video-js vjs-default-skin" controls preload="auto" width="640" height="268">
    <source src="https://vjs.zencdn.net/v/oceans-will-404.mp4" type='video/mp4'>
  </video>
  <script src="https://vjs.zencdn.net/7.2/video.js"></script>
  <script>
    var player = videojs('my_video_1', {}, function() {    
      this.on('error', function() {
        this.errorDisplay.close();
        console.log('is error display open? ' + (this.errorDisplay.opened() ? 'yes' : 'no'));
      });
    });
  </script>
```

Is working correctly:

1. When the player is ready, an `error` listener is bound.
2. The player attempts to load `oceans-will-404.mp4`, which does not exist.
3. An `error` event is triggered.
4. The first listener that is called for this event is the listener that opens the modal.
5. The listener that was added in the ready callback is called, which closes the modal.

If you look at [this example](https://jsbin.com/tihuxag/edit?html,console,output) you can see the whole workflow.

There must be something else happening that's causing the error you're seeing.

Another way you could suppress the `errorDisplay` entirely is to remove it from the player:

```js
var player = videojs('my_video_1', {});

// This does not need to wait for player to be ready.
player.errorDisplay.dispose();
```

This might be the better approach because the player will be doing less work on error events.

**Comment by gkatsev** [comment]
You can also set `errorDisplay: false` as an option of the player to disable it. Also, the recommended way to turn off the error and the display is to call `player.error(null)`.

**Comment by qbadev** [comment]
I tested Your example and it simply isn't working at all. It shows only 'player ready' and does not trigger error event.
Tried this in Safe Mode in Firefox, Chrome (Ubuntu).
Then I switched to Win7 x64 and tested in Chrome, Firefox and IE. Results are inconsistent:
it always works fine in IE 10/10
Chrome 5/10
Firefox 2/10
It seems like network and native browser video handling is an issue. I did plenty of tests and found out, that it is random. I think that videojs can't handle error event if browser triggers error event earlier.


Already solved the problem by removing `<source>` from `<video>` before `videojs` init, and adding `preload="false"` to `<video>` tag.
```
var source = {src:$('#my_video_1 source').attr('src'), type:$('#my_video_1 source').attr('type')};
$('#my_video_1 source').remove();
var player = videojs('my_video_1', {}, function(){
  console.log('ready');
  this.on('error', function(){
    console.log('error');
  });
  player.src(source);
});
```
Now, browser does not process the video tag. All event bindings are done before video metadata is loaded. All event are now handled correct.
